### ISSUE TITLE
API and log do not work

### ISSUE THREAD

**Issue by kr-deps** [Lolliedieb/lolMiner-releases#47]
With `--apiport 8080` the miner almost always crashes like this. Various ports tested.

Also `--logfile` does not seem to work. Windows 10.

```
./lolMiner.exe --algo BEAM-III --pool beam.sunpool.top:3334 --user 32f2e8765c2e8f5ea41becc5f397024c94d80cc5fc50ee917af23b260ecb3a5f.testRun --apiport 8080
+---------------------------------------------------------+
|     _       _ __  __ _                   _   ___  _     |
|    | | ___ | |  \/  (_)_ __   ___ _ __  / | / _ \/ |    |
|    | |/ _ \| | |\/| | | '_ \ / _ \ '__| | || | | | |    |
|    | | (_) | | |  | | | | | |  __/ |    | || |_| | |    |
|    |_|\___/|_|_|  |_|_|_| |_|\___|_|    |_(_)___/|_|    |
|                                                         |
|              This software is for mining                |
|              Equihash 144/5, 192/7, 210/9               |
|              BeamHash I  (EquihashR 150/5/0)            |
|              BeamHash II (EquihashR 150/5/3)            |
|              BeamHash III                               |
|              ZelHash     (EquihashR 125/4/0)            |
|              CuckarooD 29                               |
|              CuckarooM 29                               |
|              Cuckaroo  30 CTX                           |
|              Cuckatoo  31/32                            |
|                                                         |
|                                                         |
|             Made by Lolliedieb, June 2020               |
+---------------------------------------------------------+

Setup Miner...
Size:
 Device 0:
    Name:    Radeon RX 580 Series
    Address: 1:0
    Vendor:  Advanced Micro Devices (AMD)
    Memory:  8192 MByte (8142 MByte free)
    Active:  true (Selected Algorithm: BeamHash III 6G)
 Device 1:
    Name:    GeForce RTX 2060
    Address: 2:0
    Vendor:  NVIDIA Corporation
    Memory:  6144 MByte
    Active:  true (Selected Algorithm: BeamHash III 6G)
 Device 2:
    Name:    ASUS Radeon RX 470 Series
    Address: 3:0
    Vendor:  Advanced Micro Devices (AMD)
    Memory:  8192 MByte (8142 MByte free)
```

Without `--apiport`:
```
./lolMiner.exe --algo BEAM-III --pool beam.sunpool.top:3334 --user 32f2e8765c2e8f5ea41becc5f397024c94d80cc5fc50ee917af23b260ecb3a5f.testRun
+---------------------------------------------------------+
|     _       _ __  __ _                   _   ___  _     |
|    | | ___ | |  \/  (_)_ __   ___ _ __  / | / _ \/ |    |
|    | |/ _ \| | |\/| | | '_ \ / _ \ '__| | || | | | |    |
|    | | (_) | | |  | | | | | |  __/ |    | || |_| | |    |
|    |_|\___/|_|_|  |_|_|_| |_|\___|_|    |_(_)___/|_|    |
|                                                         |
|              This software is for mining                |
|              Equihash 144/5, 192/7, 210/9               |
|              BeamHash I  (EquihashR 150/5/0)            |
|              BeamHash II (EquihashR 150/5/3)            |
|              BeamHash III                               |
|              ZelHash     (EquihashR 125/4/0)            |
|              CuckarooD 29                               |
|              CuckarooM 29                               |
|              Cuckaroo  30 CTX                           |
|              Cuckatoo  31/32                            |
|                                                         |
|                                                         |
|             Made by Lolliedieb, June 2020               |
+---------------------------------------------------------+

Setup Miner...
Size:
 Device 0:
    Name:    Radeon RX 580 Series
    Address: 1:0
    Vendor:  Advanced Micro Devices (AMD)
    Memory:  8192 MByte (8142 MByte free)
    Active:  true (Selected Algorithm: BeamHash III 6G)
 Device 1:
    Name:    GeForce RTX 2060
    Address: 2:0
    Vendor:  NVIDIA Corporation
    Memory:  6144 MByte
    Active:  true (Selected Algorithm: BeamHash III 6G)
 Device 2:
    Name:    ASUS Radeon RX 470 Series
    Address: 3:0
    Vendor:  Advanced Micro Devices (AMD)
    Memory:  8192 MByte (8142 MByte free)
    Active:  true (Selected Algorithm: BeamHash III 6G)
 Device 3:
    Name:    GeForce GTX 1080
    Address: 8:0
    Vendor:  NVIDIA Corporation
    Memory:  8192 MByte
    Active:  true (Selected Algorithm: BeamHash III 6G)
 Device 4:
    Name:    Radeon RX Vega
    Address: 13:0
    Vendor:  Advanced Micro Devices (AMD)
    Memory:  8176 MByte (8112 MByte free)
    Active:  true (Selected Algorithm: BeamHash III 6G)
 Device 5:
    Name:    Intel(R) HD Graphics 610
    Address: 4294967295:4294967295
    Vendor:  Intel(R) Corporation
    Memory:  3232 MByte
    Active:  false (not selected)

Connecting to pool...
Connected to beam.sunpool.top:3334 (TLS enabled)
TLS Handshake success
Authorized worker: 32f2e8765c2e8f5ea41becc5f397024c94d80cc5fc50ee917af23b260ecb3a5f.testRun
New job received: 79129 Difficulty 1024
Start Mining...
GPU 4: Submitting share
Share accepted (21 ms)
```

But there is a 3-second lag before ` Device 3:  Name:    GeForce GTX 1080` is printed as well. But no crash.

**Comment by kr-deps** [comment]
Removed 1080 from the rig. WIth `--apiport`, it lags then crashes after `Device 1` anyway.

**Comment by Lolliedieb** [comment]
Can not reproduce the problem with api ... working in Windows and Linux for me.
But can confirm problems with --log ... this is fixed with 1.02.
